plugins/src/webrtcsink/imp.rs

```diff
@@ -189,11 +194,30 @@ struct State {
     audio_serial: u32,
     video_serial: u32,
     streams: HashMap<String, InputStream>,
+    navigation_handler: Option<NavigationEventHandler>,
 }
 
+fn create_navigation_event(sink: gst_video::Navigation, msg: &str) {
```

Comment:
```suggestion
fn create_navigation_event<IsA<N: gst_video::Navigation>(sink: &N, msg: &str) {
```

or at least

```suggestion
fn create_navigation_event(sink: &gst_video::Navigation, msg: &str) {
```

The former should avoid the `upcast()` below, for the latter you'd need to do `upcast_ref()` instead. In either case, less destructive :)